### PR TITLE
Codechange: replace seprintf with fmt::format for crashlog generation

### PR DESCRIPTION
**Pull request by rubidium42**
## Motivation / Problem

C-style string formatting vs C++-style string formatting.

## Description

Replace seprintf with fmt::format.

## Limitations

Not tested on other platforms than Unix yet.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/3rdparty/fmt/chrono.h [hunk 1]
```diff
@@ -0,0 +1,2267 @@
...
+      using namespace fmt::detail;
+      return fallback(localtime_s(&tm_, &time_));
+    }
+
+    bool fallback(int res) { return res == 0; }
+
+#if !FMT_MSC_VERSION
+    bool fallback(detail::null<>) {
+      using namespace fmt::detail;
+      std::tm* tm = std::localtime(&time_);
```
[the remark this answers is not in the document]
Comment (rubidium42) [reply src/3rdparty/fmt/chrono.h]: The code itself is ingenious. If starts with the (safe) `_r` variant. If that does not exist it tries the `_s` variant before finally falling back to the unsafe variant. It looks like it's not optimized out, so CodeQL trips over it. A test (i.e. adding printfs and then calling it) shows that the `_r` variant gets called.

File: src/3rdparty/fmt/chrono.h [hunk 2]
```diff
@@ -0,0 +1,2267 @@
...
+    bool handle(detail::null<>) {
+      using namespace fmt::detail;
+      return fallback(gmtime_s(&tm_, &time_));
+    }
+
+    bool fallback(int res) { return res == 0; }
+
+#if !FMT_MSC_VERSION
+    bool fallback(detail::null<>) {
+      std::tm* tm = std::gmtime(&time_);
```
[the remark this answers is not in the document]
Comment (rubidium42) [reply src/3rdparty/fmt/chrono.h]: The code itself is ingenious. If starts with the (safe) `_r` variant. If that does not exist it tries the `_s` variant before finally falling back to the unsafe variant. It looks like it's not optimized out, so CodeQL trips over it. A test (i.e. adding printfs and then calling it) shows that the `_r` variant gets called.

File: src/3rdparty/fmt/chrono.h [hunk 3]
```diff
@@ -0,0 +1,2267 @@
...
+      case 'z':
+        handler.on_utc_offset(numeric_system::alternative);
+        break;
+      default:
+        FMT_THROW(format_error("invalid format"));
+      }
+      break;
+    default:
+      FMT_THROW(format_error("invalid format"));
+    }
```
[the remark this answers is not in the document]
Comment (rubidium42) [reply src/3rdparty/fmt/chrono.h]: Mostly irrelevant as we have many long switch cases in our code. Furthermore it's in an imported library that will get replaced by C++20 STL at some point in any case.

File: src/os/windows/crashlog_win.cpp [hunk 4]
```diff
@@ -454,30 +430,30 @@ char *CrashLogWindows::AppendDecodedStacktrace(char *buffer, const char *last) c
 			}
 
 			/* Print module and instruction pointer. */
-			buffer += seprintf(buffer, last, "[%02d] %-20s " PRINTF_PTR, num, mod_name, frame.AddrPC.Offset);
+			fmt::format_to(output_iterator, "[{:02}] %{:20s} {}", num, mod_name, frame.AddrPC.Offset);
```
Comment (glx22) [inline src/os/windows/crashlog_win.cpp]:
```suggestion
			fmt::format_to(output_iterator, "[{:02}] {:20s} {}", num, mod_name, frame.AddrPC.Offset);
```
and 3rd param used to be displayed in hexadecimal.

New format:
`[00] %openttd              140700563567662 MainWindow::OnHotkey + 478 (D:\developpement\GitHub\glx22\OpenTTD\src\main_gui.cpp:333)`
Old format:
`[00] openttd              0x00007FF6E08300FE MainWindow::OnHotkey + 478 (D:\developpement\GitHub\glx22\OpenTTD\src\main_gui.cpp:333)`
(removing 0x and leading 0 should be fine)

File: src/os/windows/crashlog_win.cpp [hunk 5]
```diff
@@ -179,16 +155,16 @@ static void GetFileInfo(DebugFileInfo *dfi, const wchar_t *filename)
...
 {
 	wchar_t buffer[MAX_PATH];
 	DebugFileInfo dfi;
 
 	GetModuleFileName(mod, buffer, MAX_PATH);
 	GetFileInfo(&dfi, buffer);
-	output += seprintf(output, last, " %-20s handle: %p size: %d crc: %.8X date: %d-%.2d-%.2d %.2d:%.2d:%.2d\n",
-		FS2OTTD(buffer).c_str(),
-		mod,
+	fmt::format_to(output_iterator, " {:20s} handle: {} size: {} crc: {:8X} date: {}-{:02}-{:02} {:02}:{:02}:{:02}\n",
```
Comment (glx22) [inline src/os/windows/crashlog_win.cpp]: Old format:
` D:\developpement\GitHub\glx22\OpenTTD\out\build\x64-debug\openttd.exe handle: 00007FF6DFB60000 size: 36075520 crc: 343219B2 date: 2023-05-24 21:06:49`

New format:
`D:\developpement\GitHub\glx22\OpenTTD\out\build\x64-debug\openttd.exe handle: 0x7ff7664f0000 size: 36110336 crc: 9251AFBA date: 2023-05-24 20:46:39`

New format doesn't look bad.

**Review by glx22 — APPROVED** [review]
Can't say for macos or unix but windows crash log is ok for me.

**Pull request merged**